USAGE.md

```diff
@@ -0,0 +1,176 @@
+## How to use it
+
+`Soldeer` is straightforward. It can work with foundry or standalone. The foundry integration uses `foundry.toml` for dependency management while the standalone uses a special toml file called `soldeer.toml`.
+
+In the following sections we will specify all the commands with `[forge]` as prefix which means that that specific command works directly from foundry.
+
+### DEPENDENCY INSTALLATION
+
+Initialize a fresh installation by using this command. It will generate either a `foundry.toml` or `soldeer.toml` file with the latest version of forge-std. This command is primarily used when you want to integrate Soldeer into your Foundry project or replace your old Foundry setup with Soldeer's setup. By using the `--clean true` argument, you can delete the old `.gitmodules` file and the `lib` directory.
+
+```bash
+[forge] soldeer init
+```
+
+#### Soldeer repository
+
+Add the dependencies by using
+
+```bash
+[forge] soldeer install <dependency_name>~<version>
+```
+
+To search if the dependency is available, visit [https://soldeer.xyz](https://soldeer.xyz).
+
+This command will download the zip file of the dependency, unzip it, install it in the `dependencies` directory.
+
+Additionally, you can install a dependency from any zip file located at any URL:
+
+```bash
+[forge] soldeer install <dependency_name>~<version> <url>
+```
+
+This command will download the zip file of the dependency from the <url>, unzip it, install it in the `dependencies` directory.
+
+#### Git repositories
+
+The syntax is `soldeer install <dependency>~<version> <git-url>`. This will clone the repository and install the dependency in the `dependencies` folder.
+
+You can also use a certain commit as a dependency
+
+```bash
+soldeer install <dependency>~<version> git:<url> <commit>
+```
+
+Some example
+
+```bash
+[forge] soldeer install test-project~v1 git@github.com:test/test.git
+[forge] soldeer install test-project~v1 git@gitlab.com:test/test.git
+```
+
+```bash
+[forge] soldeer install test-project~v1 https://github.com/test/test.git
+[forge] soldeer install test-project~v1 https://gitlab.com/test/test.git
+```
+
+Or using custom commit hashes
+
+```bash
+[forge] soldeer install test-project~v1 git@github.com:test/test.git --rev 345e611cd84bfb4e62c583fa1886c1928bc1a464
+```
+
+```bash
+[forge] soldeer install
+```
+
+This command will install all the dependencies from the `soldeer.toml`/`foundry.toml` file.
+
+### How to push a new dependency to the repository
+
+In order to push a new dependency to the repository you have create an account on [https://soldeer.xyz](https://soldeer.xyz), create a project that it will match the dependency name.
+
+Example:
+Create a project called `my-project` and then use the `[forge] soldeer push my-project~v1.0`. This will push the project to the repository and it will be available for everyone to use.
```

Comment:
```suggestion
Create a project called `my-project` and then use the `[forge] soldeer push my-project~1.0.0`. This will push the project to the repository and it will be available for everyone to use.
```

Not sure we should recommend to use a `v` prefix for the version, as that's not the norm in the current packages.